### ISSUE TITLE
NodeBB blocked by LibreJS because of missing license information

### ISSUE THREAD

**Issue by GitMensch** [NodeBB/NodeBB#3267]
Please see http://www.gnu.org/software/librejs/free-your-javascript.html
As nodeBB is promoted as GPLv3+ it seems that adding the license information is the only necessary part (for minimized source `@source` is needed within a comment).

**Comment by GitMensch** [comment]
# Result of LibreJS: #

## List of blocked JavaScript in https://community.nodebb.org/ ##

* Whitelist
     This script is detected as inline, nonfree, defining functions or methods, and the rest of the page as loading external scripts

    		var RELATIVE_PATH = "";
    		var config = JSON.parse('{"relative_path":"","socketioTransports":["polling","websocket"],"websocketAddress":"","version":"0.7.1-dev","siteTitle":"NodeBB","showSiteTitle":false,"postDelay":"10","minimumTitleLength":"3","maximumTitleLength":"255","minimumPostLength":"8","maximumPostLength":"32767","hasImageUploadPlugin":true,"maximumProfileImageSize":"256","minimumUsernameLength":"2","maximumUsernameLength":"16","minimumPasswordLength":"6","maximumSignatureLength":"255","maximumAboutMeLength":"500","useOutgoingLinksPage":false,"allowGuestSearching":false,"allowGuestUserSearching":false,"allowGuestHandles":false,"allowFileUploads":false,"allowProfileImageUploads":true,"allowTopicsThumbnail":false,"allowAccountDelete":true,"privateUserInfo":false,"privateTagListing":false,"usePagination":false,"disableSocialButtons":false,"disableChat":false,"maxReconnectionAttempts":5,"reconnectionDelay":200,"tagsPerTopic":4,"minimumTagLength":3,"maximumTagLength":24,"topicsP…

    * Whitelist

    This script is detected as inline, nonfree, defining functions or methods, and the rest of the page as loading external scripts

    	require.config({
    		baseUrl: "/src/modules",
    		waitSeconds: 3,
    		urlArgs: "v0.5.0-2-3488-ga31178e",
    		paths: {
    			'forum': '../client',
    			'vendor': '../../vendor',
    			'mousetrap': '../../bower/mousetrap/mousetrap'
    		}
    	});```

    * Whitelist

    This script is detected as inline, nonfree, defining functions or methods, and the rest of the page as loading external scripts

    		require(['forum/footer']);```
    	

    * Whitelisterror parsing: ReferenceError: type is not defined
    https://community.nodebb.org/nodebb.min.js?abf34a6c

## List of accepted JavaScript in https://community.nodebb.org/ ##

    * LibreJS did not allow the execution of any scripts on this page: '
         * There may be no scripts on this page (check source, C-u)
         * The inline and on-page JavaScript code may not be free and/or may not have proper license information and external scripts (if present) may have been removed by default.
         * External scripts may not be free and/or may not have proper licensing and are not part of the whitelist of free JavaScript libraries.

**Comment by psychobunny** [comment]
I'm actually not too sure what to do about this. Could you perhaps make the change in the form of a pull request and test if it passes, if it's not a big change I'd be happy to accept

**Comment by GitMensch** [comment]
The first thing is to fix the ReferenceError, something that sounds like a good idea in general.

The next thing is to read the site I've linked to where the procedures of adding a parseable licence comment is described.

I think I'm currently not familiar enough with nodeBB (used it as a board twice and had to temporarily disable LibreJS to use it - it blocks every Javascript which is not detected as being free software) to fix the ReferenceError and to know which sources are included (and where the original and minified versions are).

**Comment by psychobunny** [comment]
Closed due to lack of interest. That said, more than happy to accept a PR. Additionally, this problem might be resolved after we move vendor libs to bower

**Comment by GitMensch** [comment]
The interest is still there, but I'm not a nodeBB dev and therefore contribute more effective to other free software.
I don't know why the move to bower should resolve this issue but I suggest to reopen it, either setting the same milestone as the move to bower or later.

**Comment by psychobunny** [comment]
@barisusakli @julianlam I'm not sure how we can resolve this. I've reopened this for now.

@GitMensch by lack of interest I meant from other users. I've just been cleaning our issue tracker because we want to use this tracker for bugs and high-usecase feature requests if possible

**Comment by GitMensch** [comment]
@psychobunny This**is** a high-usecase feature request / bug for a growing group of LibreJS users - which won't allow any of the Javascript parts as they are now and therefore break nodeBB :-)

**Comment by barisusakli** [comment]
@psychobunny it requires adding the license into the head in a script tag before any inline js

```
<script>
 /*
        @licstart  The following is the entire license notice for the
        JavaScript code in this page.
        
        //copy paste license
    
        @licend  The above is the entire license notice
        for the JavaScript code in this page.
*/
</script>
```

The above gets rid of the librejs warnings.

The reference error is probably a bug in librejs itself because the page doesn't have any errors when loaded in chrome or firefox. https://lists.gnu.org/archive/html/help-librejs/2015-07/msg00003.html

Although I am not convinced we should do it.

**Comment by psychobunny** [comment]
Going to have to ask OP to garner some support in our community, re-closing because apart from OP nobody's really brought it up. Also want to start using this tracker for bugs more than feature requests (see community for latter)
